src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -46,12 +44,14 @@
     private final BigQueryPdao bigQueryPdao;
     private final RestHighLevelClient client;
 
+    @Value("${elasticsearch.numShards}")
+    private static int NUM_SHARDS;
+
     @Autowired
-    public SearchService(BigQueryPdao bigQueryPdao) {
-        // needs to be moved into own class with config
-        final RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200));
+    public SearchService(BigQueryPdao bigQueryPdao, RestHighLevelClient client) {
         this.bigQueryPdao = bigQueryPdao;
-        this.client = new RestHighLevelClient(builder);
+        // injected from ElasticSearchRestClientConfigurations.java
+        this.client = client;
```

Comment:
How does it know it should use this?

Reply:
ElasticSearchRestClientConfigurations.java is a spring boot autoconfigured class, 
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.developing-auto-configuration

So it somehow ends up in the classpath and spring can find it.